electron_app/DownloadManager.js

```diff
@@ -2,22 +2,29 @@ const mtd = require('zeltice-mt-downloader');
 const path = require('path');
 const os = require('os');
 
-const handleFolder = require('./helpers/handleDisk');
+const { handleFolder, handleDiskSpace } = require('./helpers/handleDisk');
 
 let downElements = [];
 
 const downloadManager = (items, access_token) => {
-  let filePath = '';
-  const options = setHeader(access_token);
-  items.forEach(item => {
-    if (item.preserveStructure) {
-      const structurePath = item.path.substring(item.path.lastIndexOf('/'), 0);
-      handleFolder(path.join(item.destination, structurePath)).then(result =>
-        processDownload(access_token, item, result)
-      );
-    } else {
-      processDownload(access_token, item, item.destination);
-    }
+  return new Promise((res, rej) => {
```

Comment:
Now that I've looked at this more, I'm not sure that this is the correct place to do this work. We want to check for the error case before the download begins, not after. The best scenario would be to report the error before the dialog is dismissed, e.g. when computing `FileModalComponent.isValid`. If not there, it should be done in `FileModalComponent.startDownload()` before it calls `transferablesGridComponent.startDownload()`. That way you can avoid adding and then removing items from the transfer list.

Reply:
All disk verification were changed to be performed separately from any download action. 
First, check disk conditions, if everything ok, download else, show error message.